src/domain/accounts/services/src/authentication_service_impl.rs

```diff
@@ -32,22 +33,20 @@ use serde::{Deserialize, Serialize};
 
 use crate::LoginPasswordAuthProvider;
 
-///////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
 
 const KAMU_JWT_ISSUER: &str = "dev.kamu";
 const KAMU_JWT_ALGORITHM: Algorithm = Algorithm::HS384;
 const EXPIRATION_TIME_SEC: usize = 24 * 60 * 60; // 1 day in seconds
 
-///////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
 
 pub struct AuthenticationServiceImpl {
     predefined_accounts_config: Arc<PredefinedAccountsConfig>,
     time_source: Arc<dyn SystemTimeSource>,
     encoding_key: EncodingKey,
     decoding_key: DecodingKey,
-    authentication_providers_by_method: HashMap<&'static str, Arc<dyn AuthenticationProvider>>,
-    login_password_auth_provider: Option<Arc<LoginPasswordAuthProvider>>,
-    account_repository: Arc<dyn AccountRepository>,
+    transaction_runner: Arc<DatabaseTransactionRunner>,
     state: tokio::sync::Mutex<State>,
 }
```

Comment:
A few things should be changed:

1) We are not strictly required to do lazy registration of predefined accounts. This could be moved to a separate "initializer" service that is only invoked from `app.py` (and it's `node` equivalent). The initialization code will be contained in it's own transaction.

2) If suggestion above works, we can get rid of `State` and `Singleton` status

3) I think at the end of refactoring, `AuthenticationServiceImpl` should not be responsible for controlling the scope of transaction. In case of web queries, there is middleware. In case of commands, there is a transaction in `app.py` around execution of commands.

Reply:
> This could be moved to a separate "initializer" service

Added `PredefinedAccountsRegistrator` -- done 

> If suggestion above works, we can get rid of State and Singleton status
> ...
> I think at the end of refactoring, AuthenticationServiceImpl should not be responsible for controlling the scope of transaction`

Done